### PR TITLE
Allow searching for items by quality name

### PR DESCRIPTION
**Pull request by nornagon**
I haven't written PHP in about 15 years so thanks for forgiving my sloppiness 🙇

Also, I don't have a PHP environment locally, so I'm not 100% sure if this code is correct. Let me know if so!

### CODE REVIEW

File: src/app/models/Item.php [hunk 1]
```diff
@@ -341,7 +341,10 @@ public function matches($text)
 
         return $this->symbol == $text ||
         stristr($this->id, $text) ||
-        stristr($this->name, $text);
+        stristr($this->name, $text) ||
+        array_filter($this->data->qualities, function($q) {
```
Comment (DanmakuDan) [inline src/app/models/Item.php]: I also have no idea what I'm doing sometimes, but it looks like ```use ($text)``` is necessary to make use of the variable inside the function. Also, q[0] appears to be enough to allow a string comparison with the search text.
```suggestion
        array_filter($this->data->qualities,
        function ($q) use ($text) {
```

File: src/app/models/Item.php [hunk 2]
```diff
@@ -341,7 +341,10 @@ public function matches($text)
...
+            return stristr($this->repo->getModel("Quality", $q[0]), $text)
```
Comment (DanmakuDan) [inline src/app/models/Item.php]:
```suggestion
            return stristr($q[0], $text);
```

**Comment by nornagon** [comment]
Thanks for the suggestions! I've applied them & edited the indentation a bit.

**Comment by nornagon** [comment]
Looks like this is deployed! :D I can't tell, but is search slower than it used to be? Does this need some optimization work?

**Comment by DanmakuDan** [comment]
I think it's a little slower, but the cache is a large file-based key-value store currently. It's around 300 MB on disk, and I don't have memcached set up to stay in memory on the shared host server.

**Pull request merged**